dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -5,6 +5,60 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   # GET /users/auth/:provider/callback
   def all
+    if should_connect_provider?
+      connect_provider
+    else
+      login
+    end
+  end
+
+  # GET /users/auth/:provider/connect
+  def connect
+    return head(:bad_request) unless current_user&.migrated? && params[:provider].present?
+
+    session[:connect_provider] = 2.minutes.from_now
+    redirect_to omniauth_authorize_path(current_user, params[:provider])
+  end
+
+  # Call GET /users/auth/:provider/connect and the callback will trigger this code path
+  def connect_provider
+    return head(:bad_request) unless can_connect_provider? # TODO: (madelynkasula) better error here?
```

Comment:
I think `bad_request` is a reasonable response, but we might want to log failures to Honeybadger with some additional information (especially for the first month or so) to help diagnose issues that are hard to reproduce in-house.